support/rdf-utils.js

```diff
@@ -0,0 +1,139 @@
+export function findTripleWithObject(triples, objectUri) {
+  return triples.find((t) => t.object === objectUri);
+}
+
+/**
+ * @param {Triple[]} triples
+ * @returns {Triple[]}
+ */
+export function dedupeTriples(triples) {
+  const seenTriples = new Set();
+  const deduped = [];
+  for (const triple of triples) {
+    if (!seenTriples.has(hashTriple(triple))) {
+      deduped.push(triple);
+      seenTriples.add(hashTriple(triple));
+    }
+  }
+  return deduped;
+}
+const DEFAULT_PREFIX_MAP = new Map([
+  ["ext", "http://mu.semte.ch/vocabularies/ext/"],
+  ["mu", "http://mu.semte.ch/vocabularies/core/"],
+  ["muSession", "http://mu.semte.ch/vocabularies/session/"],
+  ["tmp", "http://mu.semte.ch/vocabularies/tmp/"],
+  ["besluit", "http://data.vlaanderen.be/ns/besluit#"],
+  ["bv", "http://data.vlaanderen.be/ns/besluitvorming#"],
+  ["mandaat", "http://data.vlaanderen.be/ns/mandaat#"],
+  ["persoon", "http://data.vlaanderen.be/ns/persoon#"],
+  ["generiek", "http://data.vlaanderen.be/ns/generiek#"],
+  ["mobiliteit", "https://data.vlaanderen.be/ns/mobiliteit#"],
+  [
+    "publicationStatus",
+    "http://mu.semte.ch/vocabularies/ext/signing/publication-status/",
+  ],
+  ["eli", "http://data.europa.eu/eli/ontology#"],
+  ["m8g", "http://data.europa.eu/m8g/"],
+  ["dct", "http://purl.org/dc/terms/"],
+  ["cpsv", "http://purl.org/vocab/cpsv#"],
+  ["dul", "http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#"],
+  ["adms", "http://www.w3.org/ns/adms#"],
+  ["person", "http://www.w3.org/ns/person#"],
+  ["org", "http://www.w3.org/ns/org#"],
+  ["prov", "http://www.w3.org/ns/prov#"],
+  ["regorg", "https://www.w3.org/ns/regorg#"],
+  ["skos", "http://www.w3.org/2004/02/skos/core#"],
+  ["foaf", "http://xmlns.com/foaf/0.1/"],
+  ["nao", "http://www.semanticdesktop.org/ontologies/2007/08/15/nao#"],
+  ["pav", "http://purl.org/pav/"],
+  ["schema", "http://schema.org/"],
+  ["rdfs", "http://www.w3.org/2000/01/rdf-schema#"],
+  ["sign", "http://mu.semte.ch/vocabularies/ext/signing/"],
+  ["lblodlg", "http://data.lblod.info/vocabularies/leidinggevenden/"],
+  ["lblodmow", "http://data.lblod.info/vocabularies/mobiliteit/"],
+  ["locn", "http://www.w3.org/ns/locn#"],
+  ["adres", "https://data.vlaanderen.be/ns/adres#"],
+  ["persoon", "http://data.vlaanderen.be/ns/persoon#"],
+  ["notulen", "http://lblod.data.gift/vocabularies/notulen/"],
+  ["nfo", "http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#"],
+  ["nie", "http://www.semanticdesktop.org/ontologies/2007/01/19/nie#"],
+  ["dbpedia", "http://dbpedia.org/ontology/"],
+  ["besluittype", "https://data.vlaanderen.be/id/concept/BesluitType/"],
+]);
+export const RDF_TYPE = "http://www.w3.org/1999/02/22-rdf-syntax-ns#type";
+/**
+ * @param {string?} uri
+ */
+export function expandURI(uri, prefixMap = DEFAULT_PREFIX_MAP) {
+  if (typeof uri !== "string") {
+    return uri;
+  }
+  if (isURI(uri)) {
+    return uri;
+  }
+  if (uri === "a") {
+    return RDF_TYPE;
+  }
+  if (/\s/.test(uri)) {
```

Comment:
I don't get this one, is there a case where spaces would exist in the uri? I guess we're not expecting it and can't handle it, but it feels weird to just ignore it. Does it make sense to log these things or would that potentially create a lot of logs? I'm wondering if we should `.trim()` also?

Reply:
yes actually
the thing is that this function doesn't just recieve uris
it also receives things that aren't uris, such as strings, which can (and do, given html strings) contain newlines

Reply:
this is the "rather hacky" part I mentioned in the PR description